### PR TITLE
afaryy/c02-network06

### PR DESCRIPTION
**Pull request by afaryy**
# Description

Exercise (e.g. C01-E01): c02-network06

Student Name:Yvonne Yao(afaryy)

> A brief explanation of what has been done.

## Type of change

- [X] Exercise Submission (one exercise per PR)
  - [X] This PR is just for **one** exercise of a class
  - [X] PR name follows the pattern `<github-username>/<exercise-number>`
  - [X] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [X] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/02class/exercises/c02-network06/afaryy/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# c02-network06
+
+
+- Create an Elastic IP:
+```bash
+export AWS_PROFILE=devopsacademyau
```
Comment (roperto) [inline classes/02class/exercises/c02-network06/afaryy/README.md:6]: Nice to see you are using a profile for that :-)

File: classes/02class/exercises/c02-network06/afaryy/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+aws sts get-caller-identity
```
Comment (roperto) [inline classes/02class/exercises/c02-network06/afaryy/README.md:7]: Why do you need this? Were you just checking if using the correct user?

File: classes/02class/exercises/c02-network06/afaryy/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+
+```
+- Creating a NAT gateway and attach it to a subnet with an Elastic IP
+```bash
+PUBLIC_SUBNET_NAME='da-c02-public-a';
+SUBNET_ID=$(aws ec2 describe-subnets  \
+    --filter "Name=tag:Name,Values=$PUBLIC_SUBNET_NAME" \
+    --query 'Subnets[].SubnetId' --output text);
```
Comment (roperto) [inline classes/02class/exercises/c02-network06/afaryy/README.md:32]: Nice usage of variables

File: classes/02class/exercises/c02-network06/afaryy/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+ECHO $SUBNET_ID
```
Comment (roperto) [inline classes/02class/exercises/c02-network06/afaryy/README.md:33]: In linux, `echo` must be lowercase. This probably works on Windows but ideally we try to use the linux commands.

The variable names like `SUBNET_ID` are fine in uppercase, only the `ECHO` should be `echo`.

File: classes/02class/exercises/c02-network06/afaryy/README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    --tag-specfications 'ResouceType=natgateway, Tags=[{Key=Name, Value=da-c02-ngw}] ' \
+    --query=NatGateway.NatGatewayId \
+    --output text )
+
+aws ec2 describe-nat-gateways  \
+    --filter "Name=nat-gateway-id,Values=$NGW_ID"
+# OR 
+aws ec2 describe-nat-gateways  \
+    --nat-gateway-ids $NGW_ID
+
```
Comment (roperto) [inline classes/02class/exercises/c02-network06/afaryy/README.md:47]: two examples, nice :-)

File: classes/02class/exercises/c02-network06/afaryy/README.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    ]
+}
+
+
+```
+- Clean when needless
+```bash
+aws ec2 delete-nat-gateway --nat-gateway-id $NGW_ID
+
+aws ec2 release-address --allocation-id $EIP_ID
```
Comment (roperto) [inline classes/02class/exercises/c02-network06/afaryy/README.md:82]: nice adding the delete commands -- the EIP will generate cost if you dont delete.

File: classes/02class/exercises/c02-network06/afaryy/README.md (hunk 2 above)
Comment (afaryy) [reply classes/02class/exercises/c02-network06/afaryy/README.md:7]: It is just a habbit to check again.

File: classes/02class/exercises/c02-network06/afaryy/README.md (hunk 3 above)
Comment (afaryy) [reply classes/02class/exercises/c02-network06/afaryy/README.md:32]: Thanks

File: classes/02class/exercises/c02-network06/afaryy/README.md (hunk 4 above)
Comment (afaryy) [reply classes/02class/exercises/c02-network06/afaryy/README.md:33]: Recently, I prefer git bash. I try to use different tools. Both ECHO and echo are ok in git bash. I don't think it is a big problem to finish the excercises by trying different tools in diffrerent OS. However, I will consider to use cygwin and avoid using ECHO later. Thanks.

File: classes/02class/exercises/c02-network06/afaryy/README.md (hunk 5 above)
Comment (afaryy) [reply classes/02class/exercises/c02-network06/afaryy/README.md:47]: Thanks

File: classes/02class/exercises/c02-network06/afaryy/README.md (hunk 1 above)
Comment (afaryy) [reply classes/02class/exercises/c02-network06/afaryy/README.md:6]: Thanks

**Comment by afaryy** [comment]
Both ECHO and echo are ok in Git bash. Thanks.

**Comment by afaryy** [comment]
Please, I don't think ECHO is a big problem. Both ECHO and echo are ok in
git bash. As to Devops, we also need to try different tools. Although I
will consider changing the tool later as your suggestion. Thanks

Daniel Thee Roperto <notifications@github.com> 于2020年7月9日周四 上午1:41写道：

**Pull request closed without merging**